### ISSUE TITLE
As a member, I want to add other members to my social media account from the DevBetter website

### ISSUE THREAD

**Issue by snowfrogdev** [DevBetterCom/DevBetterWeb#125]
Integrate various social media accounts to devBetter so that members can see which members they are not already friends with and easily add them without having to leave the site.

**Comment by ardalis** [comment]
This sounds cool. I assume it would only work if you were already/still logged into whichever account while on dB's site.

**Comment by snowfrogdev** [comment]
I'm going to try my hand at this but for this issue I will restrict the scope to only Twitter. We can make additional issues for other social media accounts.

**Comment by snowfrogdev** [comment]
What do you think about this little animated icon to signify that you haven't connected to a member's social account yet?

![Animation](https://user-images.githubusercontent.com/22530235/187045948-97b5a9a2-d742-4078-8624-e46fa91e2c16.gif)

**Comment by ardalis** [comment]
Sure, looks good.

**Comment by snowfrogdev** [comment]
After some prototyping with the Twitter API, I conclude that this feature would be non-trivial to implement for Twitter. Two main reasons for that:

1) Twitter does not provide an official C# SDK. The community maintained SDKs are poorly maintained and are not up to date with the latest version of Twitter's API.
2) Twitter has severe rate limitation on most of it's API endpoints making it impossible to provide the functionality we hoped for in real-time. For this to work we would need to implement complex caching and job queuing.

The feature still has merit but I suggest we create separate issues for each of the social account we have listed. We may even make one for Twitter despite my findings. Someone with more bandwidth may still decide to pick it up. I may even choose to do so in the future.

I've researched it quickly and here are the organizations that seem to offer an API making it possible to implement this feature in part or in full:
- Twitter
- Github
- Twitch
- Youtube

We can probably make an issue for each one.

As a reminder, for this feature to make sense, there really is two requirements:
1) A devBetter member should be able to see, in the Members list, if they follow a member or not.
2) A devBetter member should be able to click a button to follow a given member that he doesn't already follow.
